python/lsst/dax/apdb/apdb.py

```diff
@@ -288,8 +290,8 @@ def containsVisitDetector(self, visit: int, detector: int) -> bool:
         Returns
         -------
         present : `bool`
-            `True` if some DiaObject, DiaSource, or DiaForcedSource records
-            exist for the specified observation, `False` otherwise.
+            `True` if at least one DiaSource or DiaForcedSource record
+            may exist for the specified observation, `False` otherwise.
```

Comment:
Why the "may" here?  Could this return `True` if someone else is simultaneously removing all the `DiaSource` or `DiaForcedSource` records?  Might "at least one record exists at the time of checking" be a better phrasing?

Reply:
One proposed implementation on the Cassandra side is to have a separate table of visit-detector combinations that would be updated before writing sources [see Jira discussion](https://rubinobs.atlassian.net/browse/DM-41670?focusedCommentId=512188). It's possible for an error to occur after that table has been updated but before any actual sources have been written.

I think such false positives are acceptable (which is why I didn't complain), but the spec should reflect the fact that (for error handling purposes) it's pessimistic. There is precedent for this kind of ambiguity in the Stack, e.g. in [DatasetQueryResults.any](https://pipelines.lsst.io/v/daily/py-api/lsst.daf.butler.registry.queries.DatasetQueryResults.html#lsst.daf.butler.registry.queries.DatasetQueryResults.any).

Reply:
So the utility of the method is that if it returns `False` you definitely know there is nothing to retrieve (yet), and if it returns `True`, you know that someone tried to write records, but you might or might not be able to retrieve any?